web/src/modals.ts

```diff
@@ -221,9 +221,9 @@ export function close_if_open(modal_id: string): void {
     }
 
     const $micromodal = $(".micromodal.modal--open");
-    const active_modal_id = CSS.escape(`${CSS.escape($micromodal.attr("id") ?? "")}`);
-    if (active_modal_id === `${CSS.escape(modal_id)}`) {
-        Micromodal.close(`${CSS.escape($micromodal.attr("id") ?? "")}`);
+    const active_modal_id = CSS.escape(CSS.escape($micromodal.attr("id") ?? ""));
```

Comment:
It doesn’t make any sense to double-CSS-escape something; there has to be something wrong here…

Reply:
Quite so. @sahil839 do you know what the story with this code is?

Reply:
The code was added in #25512 by Aman. We do need escaping here as suggested by Anders.

Reply:
Huh? When did I suggest that we need escaping here, let alone double-escaping?

Escaping is not some magic fairy dust that you sprinkle everywhere to make your code pretty. It has a specific purpose. Its purpose is to correctly insert an inner string into an outer string in some language, such that, when the outer string is later parsed in that language, the inner string will be recovered as-is and won’t be reinterpreted as extra syntax in the outer language.

For example, if the outer language is CSS selectors, we have

```js
» id = "foo.bar"
» selector = `#${id}`
"#foo.bar"  // wrong: selects element with id `foo`, class `bar`
» selector = `#${CSS.escape(id)}` 
"#foo\\.bar"  // right: selects element with id `foo.bar`
```

But `active_modal_id` is not a CSS selector. It is not an outer language that’s going to be re-parsed to recover an inner string. Escaping is incorrect here.

And double-escaping is wrong even in the CSS selector case:

```js
» selector = `#${CSS.escape(CSS.escape(id))}`
"#foo\\\\\\.bar"  // wrong: selects element with id `foo\.bar`
```

The only time when a string should be escaped twice is when it’s going to be parsed twice in a nested way (for example, a URL query parameter in an HTML attribute needs to be URL-encoded and then HTML-escaped so that it can be correctly HTML-parsed and then URL-parsed).

Reply:
Oops, very sorry. I actually meant to write "do not need", should have read the comment once after posting it.
Again sorry for the inconvenience.